genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -98,12 +120,27 @@ def __init__(self, rigid_solver):
             sdist=gs.ti_float,
         )
         self.simplex_vertex = struct_simplex_vertex.field(shape=(self._B, 4))
+        self.simplex_buffer = struct_simplex_buffer.field(shape=(self._B, 4))
         self.simplex = struct_simplex.field(shape=(self._B,))
-        self.simplex_vertex_intersect = struct_simplex_vertex.field(shape=(self._B, 4))
-        self.simplex_buffer_intersect = struct_simplex_buffer.field(shape=(self._B, 4))
-        self.nsimplex = ti.field(dtype=gs.ti_int, shape=(self._B,))
 
-        ### EPA
+        # Only when we enable MuJoCo compatibility, we use the simplex vertex and buffer for intersection checks.
+        if self._enable_mujoco_compatibility:
+            self.simplex_vertex_intersect = struct_simplex_vertex.field(shape=(self._B, 4))
+            self.simplex_buffer_intersect = struct_simplex_buffer.field(shape=(self._B, 4))
+            self.nsimplex = ti.field(dtype=gs.ti_int, shape=(self._B,))
+
+        # In safe GJK, we do not allow degenerate simplex to happen, because it becomes the main reason of EPA errors.
+        # To prevent degeneracy, we throw away the simplex that has smaller degeneracy measure (e.g. colinearity,
+        # coplanarity) than this threshold.
+        self.simplex_max_degeneracy = gs.np_float(1e-5)
```

Comment:
This variable is only used squared, and within the scope of taichi kernels. You should probably defined `self.simplex_max_degeneracy_sq = gs.ti_float(1e-5) ** 2` (or directly `gs.ti_float(1e-10)`) instead. But overall, I don't think wrapping constant scalar with `gs.np_(float|int|...)` is any useful.